server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java

```diff
@@ -432,4 +432,10 @@ Table buildTable(RestRequest request, ClusterStateResponse state, IndicesStatsRe
 
         return table;
     }
+
+    // Remove sync_id from the API in V10?
+    @UpdateForV10(owner = UpdateForV10.Owner.DISTRIBUTED_INDEXING)
+    private static void addSyncId(Table table) {
```

Comment:
We can remove this field from the API output without making a breaking change. Does it make sense to return conditionally only for 8.x based on `RestVersion`?